source/extensions/filters/network/redis_proxy/conn_pool_impl.cc

```diff
@@ -379,7 +419,9 @@ void InstanceImpl::PendingRequest::onResponse(Common::Redis::RespValuePtr&& resp
 void InstanceImpl::PendingRequest::onFailure() {
   request_handler_ = nullptr;
   pool_callbacks_.onFailure();
-  parent_.parent_.onFailure();
+  if (auto shared_conn_pool = parent_.parent_.lock()) {
```

Comment:
Same comment here, if the filter goes away we should be cancelling things. I don't think these are all needed.